activerecord/lib/active_record/fixtures.rb

```diff
@@ -977,10 +978,24 @@ def setup_fixtures(config = ActiveRecord::Base)
           @loaded_fixtures = load_fixtures(config)
           @@already_loaded_fixtures[self.class] = @loaded_fixtures
         end
+
+        # Begin transactions for connections already established
         @fixture_connections = enlist_fixture_connections
         @fixture_connections.each do |connection|
           connection.begin_transaction joinable: false
         end
+
+        # When connections are established in the future, begin a transaction too
+        @connection_subscriber = ActiveSupport::Notifications.subscribe('connection.active_record') do |_, _, _, _, payload|
+          if payload[:class_name] && klass = payload[:class_name].constantize rescue nil
```

Comment:
@arthurnn Fixed and removed the assignment-in-conditional which I think is also discouraged in Rails style guide.